pyfr/plugins/base.py

```diff
@@ -62,6 +62,26 @@ def serialise(self, intg):
         return {}
 
 
+class BaseCLIPlugin:
+    name = None
+    systems = None
+    formulations = None
+    cli = None
+
+    @classmethod
+    def parse_cli(cls, parser):
```

Comment:
I'd force subclasses to implement this.  Makes the code more explicit and makes it easier for them to set a process method.